### DISCUSSION TITLE
Update a repo with the new version number of a private JS repo

### DISCUSSION THREAD

**Discussion by James-Hal** [renovatebot/renovate#28043] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

I am working on a javascript project in gitlab where I want to update the version number of a package pulled from a private repo in the package.json file and also update the package-lock.json file, I can update the version number in the package file using a custom manager but this does not update the lock file and I cannot see a way to use the npm manager to update this version number either.  Would it be possible to give the custom manager an option to allow it to use the npm manager’s functions to update the package-lock file?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Can you clarify why you can't use the npm manager in the first place? Normally if you say "I want to update the version number of a package pulled from a private repo in the package.json file" then that's an obvious use of the npm manager

**Reply by James-Hal** [reply]
Hi, I couldn't see something for that in the docs, I found an option for extending managers to look at different files but as it's in the same file I didn't think that would help,

**Reply by rarkins** [reply]
What I mean is that you have described the behavior of the npm manager. Please expand your description to describe why it's not. I literally do not understand why you cannot use npm in the first place, and your answer here didn't help

**Reply by James-Hal** [reply]
I am using the NPM manager and it is updating all of the other dependencies but not the package from the private repo

**Reply by rarkins** [reply]
Then is your problem related to authentication?
